Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -1604,6 +1605,44 @@ public void execution_deleteContext(@NotNull final String contextId) {
     }
   }
 
+  @Nullable
+  public RuntimeCompletionResult execution_getSuggestions(@NotNull final String code,
+                                                          final int offset,
+                                                          @NotNull final VirtualFile contextFile,
+                                                          final int contextOffset,
+                                                          @NotNull final List<RuntimeCompletionVariable> variables,
+                                                          @NotNull final List<RuntimeCompletionExpression> expressions) {
+    final String contextFilePath = FileUtil.toSystemDependentName(contextFile.getPath());
+
+    final AnalysisServer server = myServer;
+    if (server == null) {
+      return new RuntimeCompletionResult(Lists.newArrayList(), Lists.newArrayList());
+    }
+
+    final CountDownLatch latch = new CountDownLatch(1);
+    final AtomicReference<RuntimeCompletionResult> refResult = new AtomicReference<>();
+    server.execution_getSuggestions(
+      code, offset,
+      contextFilePath, contextOffset,
+      variables, expressions,
+      new GetRuntimeCompletionConsumer() {
+        @Override
+        public void computedResult(RuntimeCompletionResult result) {
+          refResult.set(result);
+          latch.countDown();
+        }
+
+        @Override
+        public void onError(RequestError error) {
+          logError("execution_getSuggestions()", contextFilePath, error);
```

Comment:
I've wrapped it with `if (!RequestErrorCode.UNKNOWN_REQUEST.equals(error.getCode())) {}` to be old SDK-compatible